### PR TITLE
G-mode: Green Hill Zone

### PR DESCRIPTION
**Pull request by kjbranch**

### CODE REVIEW

File: region/brinstar/green.json [hunk 1]
```diff
@@ -4177,6 +4291,98 @@
...
+                    "Use a Power Bombs to kill the first three hoppers,",
+                    "or very carefully dodge them by quickly rolling between the first two tall flowers and waiting for the hopper to jump over Samus.",
+                    "The middle hopper can be killed safely with many Bombs by staying on the top left of the raised ground,",
+                    "without overhanging or the hopper can hit Samus. Alternatively, lure and spring ball jump past it.",
+                    "Killing the bugs in their pipes is possible without getting hit, or use spring ball to jump past them.",
+                    "Once the bugs are killed once they will stop respawning.",
+                    "The last hopper can be carefully killed with Bombs, or killed with a Power Bomb.",
+                    "IBJ or bomb boost spring ball jump up to the top right door."
+                  ],
+                  "devNote": "The canTrickyJump is used for rolling into a precise location, dodging hoppers while rolling, and spring ball jumping past enemies."
```
Comment (osse101) [inline region/brinstar/green.json:4343]: I found dodging the hoppers to be easier than killing them with bombs.  Takes some practice either way, to learn enemy patterns and how to deal with pipe bugs.  The top hopper I dealt with by letting it jump off to the right.

File: region/brinstar/green.json [hunk 2]
```diff
@@ -4177,6 +4291,98 @@
...
+                      "canTrickyJump",
+                      {"and": [
+                        "Bombs",
+                        {"ammo": { "type": "PowerBomb", "count": 1 }}
+                      ]},
+                      {"ammo": { "type": "PowerBomb", "count": 3 }}
+                    ]},
+                    {"or": [
+                      "h_canArtificialMorphIBJ",
+                      "h_canArtificialMorphSpringBallBombJump"
```
Comment (osse101) [inline region/brinstar/green.json:4330]: I needed 2 springball bomb jumps.  One to reach the the ledge with the bomb blocks, and another to get up to the door.

**Comment by osse101** [comment]
With Hijump you need to turn off springball for a SpringFling to reach the door or item, neat.

**Comment by kjbranch** [comment]
HiJump to the top right door doesnt need g-mode. 
I can't get to the pipe with hijump without a bomb boost. A bomb boost can get to the pipe with or without it, it just needs to be a horizontal boost with it.

**Comment by kjbranch** [comment]
Wow I get it now. That's really cool!

**Comment by kjbranch** [comment]
I figure it would be best to create the spring fling tech in the next PR, and try to use it in some other places as well.

**Pull request merged**